### PR TITLE
LPS-185693 Allow to configure Deep Search settings at instance level

### PR DESCRIPTION
**Pull request by felipelorenz**
https://liferay.atlassian.net/browse/LPS-185693

### CODE REVIEW

**Comment by felipelorenz** [comment]
ci:test:relevant

File: modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/deep/pagination/configuration/DeepPaginationConfigurationWrapper.java [hunk 1]
```diff
@@ -32,7 +32,28 @@
...
+
+			return _configurationProvider.getSystemConfiguration(
+				DeepPaginationConfiguration.class);
+		}
+		catch (ConfigurationException configurationException) {
+			return ReflectionUtil.throwException(configurationException);
+		}
+	}
+
+	public boolean getEnableDeepPagination(long companyId) {
```
Comment (peerkar) [inline modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/deep/pagination/configuration/DeepPaginationConfigurationWrapper.java:54]: I would renamce to enableDeepPagination or maybe to more 'standard' isEnableDeepPagination. 'get' is usually not used to 'get' booleans.

Taking a look at https://github.com/liferay/liferay-portal/blob/99262f8fa6714886eca405ef5e0e5c407327e194/modules/apps/portal-search-elasticsearch7/portal-search-elasticsearch7-impl/src/main/java/com/liferay/portal/search/elasticsearch7/internal/configuration/ElasticsearchConfigurationWrapper.java#L61-L67 would speak for 'enableDeepPagination'

**Comment by peerkar** [comment]
@felipelorenz just a minor thing, but I think Brian might appreciate generally if you could squash the commits to the same thing to a single commit. Here I mean getting rid of the 3rd one. No always avoidable, but generally. 

Matter of taste, but I would probably split the second commit d5a508332ed80dbab3c6d8318247dc98a262fa26 into two: one for the configurationwrapper and the other for the searcher

Other than that, please rename the getter for boolean and looks good to go 👍

**Comment by felipelorenz** [comment]
ci:test:sf

**Comment by felipelorenz** [comment]
ci:test:relevant

**Comment by felipelorenz** [comment]
ci:test:sf

**Comment by felipelorenz** [comment]
ci:test:search

**Comment by felipelorenz** [comment]
ci:test:relevant

**Comment by felipelorenz** [comment]
ci:reevaluate:1267401_10223

**Comment by felipelorenz** [comment]
ci:forward:force

**Pull request closed without merging**